hack/gomod.sh

```diff
@@ -5,44 +5,63 @@
 #
 # NOTE: This won't work unless the build environment has internet access
 
-set -ux
+set -eux
 
-IS_CONTAINER=${IS_CONTAINER:-false}
+IS_CONTAINER="${IS_CONTAINER:-false}"
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+MOUNTDIR="/workdir"
+WORKDIR="/tmp${MOUNTDIR}"
 
 if [ "${IS_CONTAINER}" != "false" ]; then
-  export XDG_CACHE_HOME="/tmp/.cache"
+    export XDG_CACHE_HOME="/tmp/.cache"
 
-  STATUS=$(git status --porcelain)
-  if [ -n "$STATUS" ]; then
-      echo "Dirty tree: refusing to continue out of caution"
-      exit 1
-  fi
+    # locally we need to make a copy for writing, while in prow we don't
+    if [ "${COPY_REPOSITORY:-}" = "TRUE" ]; then
+        cp -a "${MOUNTDIR}"/ "${WORKDIR}"/
+        # newer git versions have safe directory mechanism we need to disable
+        git config --global --add safe.directory "${WORKDIR}"
+        # move workdir to the repository copy
+        cd "${WORKDIR}" || { echo "error: can't enter ${WORKDIR}"; exit 1; }
+    fi
 
-  go mod tidy
-  rc=$?
+    STATUS="$(git status --porcelain)"
+    if [ -n "${STATUS}" ]; then
+        echo "Dirty tree: refusing to continue out of caution"
+        exit 1
+    fi
 
-  if [ $rc -ne 0 ]; then
-      echo "'go mod tidy' failed"
-      exit 1;
-  fi
+    # sh doesn't allow ** recursive glob
+    for mod_file in *.mod */*.mod */*/*.mod; do
```

Comment:
Have you thought of a taking advantage of using https://github.com/metal3-io/ip-address-manager/blob/main/Makefile#L160-L164 target ?

Reply:
I was thinking the same also. In CAPM3 we use make targets in these scripts (e.g. [here](https://github.com/metal3-io/cluster-api-provider-metal3/blob/b640d55d7a3027c6b76bc3affd2c05b6b51eacce/hack/unit.sh#L13)). The glob here would obviously handle it nicely if we add more modules, but we could easily forget to update the Makefile.

Reply:
Good point, I missed the fact the make actually works in the golang image. Converted to use the make target now.